lib/SPIRV/SPIRVLowerBool.cpp

```diff
@@ -101,8 +101,12 @@ class SPIRVLowerBool : public ModulePass, public InstVisitor<SPIRVLowerBool> {
   }
   virtual void visitUIToFPInst(UIToFPInst &I) {
     auto Op = I.getOperand(0);
-    if (isBoolType(Op->getType())) {
-      auto Ty = Type::getInt32Ty(*Context);
+    auto OpTy = Op->getType();
+    if (isBoolType(OpTy)) {
+      auto *Ty = static_cast<Type *>(Type::getInt32Ty(*Context));
```

Comment:
You should be able to avoid the `static_cast` by not relying on `auto`:
```suggestion
      Type *Ty = Type::getInt32Ty(*Context);
```